WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -52,13 +53,11 @@ public bool TryGetExactMatch(Money target, [NotNullWhen(true)] out List<Money>?
 			}
 		}
 
-		public Money CalcEffectiveValue(List<Money>? list)
+		public Money CalcEffectiveValue(List<Money> list)
 		{
+			Guard.NotNull(nameof(list), list);
```

Comment:
You don't really need this. https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references (NRT) exist for the purpose that you know and your IDE tells you that there is some issue regarding nullability. 

Obviously, you can put it here if you like but I guess it was more useful before NRT were introduced in this codebase.

Reply:
IMHO this looks better than checking `if (list is null) { return 0 }`.

Reply:
I would simply remove L58 and not replace it with anything else. The purpose of NRT is to tell the reader what you should pass as an argument. If you pass deliberately `null` even when a method parameter is marked as non-`null`, then it's a bug and IDE warns you so ...